suitcase/fields.py

```diff
@@ -831,6 +832,12 @@ def unpack(self, data, **kwargs):
         self._value = self.substructure()
         return self._value.unpack(data, **kwargs)
 
+    def setval(self, value):
+        if isinstance(value, suitcase.structure.Structure):
```

Comment:
Ok, I was thinking slowly.  I see what you are trying to do here.

I've been thinking for awhile about having a way to instantiate new Structures without having to assign to each field individually.  I think that could help with this problem as well.  This would look something like this:

```python
# Arguments for creating a new structure with fields initialized
Structure(*positional_field_values, **field_values_by_key)

# Assign to members of existing structure in memory
structure.set(*positional_field_values, **field_values_by_key)
```

The `.set` variant could be used in this context.  I don't think overloading the normal assignment operator makes sense as many structures have multiple fields.

Does this sound like an acceptable approach?